meson.build

```diff
@@ -94,6 +95,16 @@ nasm_system_flags = [
 default_c_flags = []
 default_cpp_flags = []
 
+add_project_arguments([
+    '-Wl,-z,relro',
+    '-Wl,-z,noexecstack',
+    '-pie',
+    '-fstack-protector-strong',
+    '-fstack-clash-protection',
```

Comment:
All additions are good. These two are the only ones I want to validate first that we don't compromise on our detection capabilities in our tests.